simulation/behavior_tree_plugin/src/vehicle/follow_trajectory_sequence/polyline_trajectory_follower.cpp

```diff
@@ -0,0 +1,647 @@
+#include <quaternion_operation/quaternion_operation.h>
+
+#include <arithmetic/floating_point/comparison.hpp>
+#include <behavior_tree_plugin/vehicle/follow_trajectory_sequence/polyline_trajectory_follower.hpp>
+#include <cmath>
+#include <geometry/vector3/hypot.hpp>
+#include <geometry/vector3/norm.hpp>
+#include <geometry/vector3/normalize.hpp>
+#include <geometry/vector3/operator.hpp>
+#include <geometry/vector3/truncate.hpp>
+#include <iostream>
+
+namespace entity_behavior
+{
+namespace vehicle
+{
+template <typename F, typename T, typename... Ts>
+auto any(F f, T && x, Ts &&... xs)
+{
+  if constexpr (math::geometry::IsLikeVector3<std::decay_t<decltype(x)>>::value) {
+    return any(f, x.x, x.y, x.z);
+  } else if constexpr (0 < sizeof...(xs)) {
+    return f(x) or any(f, std::forward<decltype(xs)>(xs)...);
+  } else {
+    return f(x);
+  }
+}
+
+auto is_infinity_or_nan = [](auto x) constexpr { return std::isinf(x) or std::isnan(x); };
+
+std::optional<traffic_simulator_msgs::msg::EntityStatus>
+PolylineTrajectoryFollower::followTrajectory(
+  const traffic_simulator_msgs::msg::EntityStatus & entity_status,
+  std::shared_ptr<traffic_simulator::follow_trajectory::Parameter<
+    traffic_simulator::follow_trajectory::Polyline>> & trajectory_parameter,
+  const traffic_simulator_msgs::msg::BehaviorParameter & behavior_parameter, double step_time,
+  std::optional<traffic_simulator_msgs::msg::VehicleParameters>
+    vehicle_parameters /* = std::nullopt */)
+{
+  if (trajectory_parameter->shape.vertices.empty()) {
+    return std::nullopt;
+  }
+
+  entity_status_m = entity_status;
+  trajectory_parameter_m = trajectory_parameter;
+  behavior_parameter_m = behavior_parameter;
+  step_time_m = step_time;
+  vehicle_parameters_m = vehicle_parameters;
+
+  auto position = getCurrentPosition();
+  auto target_position = getTargetPosition();
+  auto front_waypoint_info = getDistanceAndTimeToFrontWaypoint(target_position, position);
+
+  if (front_waypoint_info == std::nullopt) {
+    discardTheFrontWaypoint();
```

Comment:
```suggestion
    discardTheFrontWaypoint(trajectory_parameter_m);
```
przekazane przez referencje, w efekcie usuniety waypoint

Reply:
wycofuje